File: AsalSayi.java
public class AsalSayi {
    public static void main(String[] args) {
        
        for (int index = 2; index < 1_000_000; index++){
            boolean sonuc  = asal_sayi_mi(index);
            if(sonuc){
                System.out.println(index);
            }
        }
        
    }

    public static boolean asal_sayi_mi(int sayi){
        int max_kontrol_sayi = (int)sayi/2;
        for (int index = 2; index < max_kontrol_sayi; index++){
            if (sayi % index == 0){
                return false;
            }
        }
        return true;
    }
}

File: asal_sayi.py
def asal_sayi_mi(sayi):
    if sayi < 2: return False
    
    max_kontrol_sayi = sayi//2                      #girilen sayının yarısına kadar kontrol etmek yeterli
    for index in range(2, max_kontrol_sayi + 1):    #2 den başlayıp sayısının yarısına kadar 
        if sayi % index == 0:
            return False

    return True 

for index in range(1, 1_000_000):
    sonuc   = asal_sayi_mi(index)
    if sonuc:
        print("{:,}".format(index))